src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -367,6 +375,17 @@ private string ResolvePackageAssetPath(LockFileTargetLibrary package, string rel
             return paths;
         }
 
+        private void CheckRuntimePackageDependencies(LockFileTarget compileTimeTarget, LockFileTarget runtimeTarget)
+        {
+            if (EnsureRuntimePackageDependencies && !string.IsNullOrEmpty(RuntimeIdentifier))
+            {
+                if (compileTimeTarget.Libraries.Count >= runtimeTarget.Libraries.Count)
```

Comment:
I know you discussed this logic with me, but it's probably a good idea to get feedback from the CoreFx and NuGet folks to make sure it will work correctly.

Reply:
cc @ericstj We discussed this approach.